web/cobrands/fixmystreet/staff.js

```diff
@@ -102,13 +102,17 @@ fixmystreet.staff_set_up = {
         var $phoneInput = $('input[name=phone]');
         var $showNameCheckbox = $('input[name=may_show_name]');
         var $addAlertCheckbox = $('#form_add_alert');
+        var $extraTitle = $('select#form_fms_extra_title');
         if (val === 'myself') {
             $emailInput.val($emailInput.prop('defaultValue')).prop('disabled', true);
             $emailOptionalLabel.addClass('hidden');
-            $nameInput.val($nameInput.prop('defaultValue')).prop('disabled', false);
-            $phoneInput.val($phoneInput.prop('defaultValue')).prop('disabled', false);
+            $nameInput.val($nameInput.prop('defaultValue')).prop('readonly', true);
```

Comment:
This change means that if the staff user switches to "another user", the field remains readonly and cannot be edited. And if you switch to anonymous user and back to yourself, the form is then both readonly and disabled. Is there a reason you switched from disabled to readonly? I can't see one, but may have missed something. It will either have to be switched back to disabled, or the other options here will need to be updated to deal with this change also. The test does not cover switching to another user and checking the fields are then editable.

Reply:
I was getting an error when using disabled because of course that means that those fields are not included in the POST. Report::New seems to require the name field. That is why I went with readonly. 

Thanks for pointing out that this is not finished! 😬